rest_framework_api_key/models.py

```diff
@@ -125,3 +154,38 @@ def __str__(self) -> str:
 
 class APIKey(AbstractAPIKey):
     pass
+
+
+class EndpointPermission(models.Model):
+    METHODS = [
+        'CONNECT',
+        'DELETE',
+        'GET',
+        'HEAD',
+        'OPTIONS',
+        'PATCH',
+        'POST',
+        'PUT',
+        'TRACE',
+    ]
+    METHOD_CHOICES = [(method, method) for method in METHODS]
+
+    path = models.CharField(
+        max_length=100,
+        help_text=('A regex matching the URL path. '
+                   'This should begin with / and include the version, eg "/v1/time"'),
+        validators=[validators.validate_regex_pattern],
+    )
+    method = models.CharField(
+        max_length=100,
+        choices=METHOD_CHOICES,
+        default='',
+    )
+
+    def __str__(self):
+        return f'{self.path}, {self.method}'
+
+    class Meta:
+        verbose_name = 'Endpoint Permission'
+        verbose_name_plural = 'Endpoint Permissions'
+        unique_together = ('path', 'method')
```

Comment:
The need for a separate model here is that we want being able to define per-view/per-method permission *for each API key* at runtime, right?

I think I can see where this is going, but the fact that we're storing path regexes in the database doesn't seem right to me: if we change the code, the content of the DB needs to change.

I've been thinking that this sort of feature before, especially in #29 which drafts a notion of **API key scopes** that somewhat replicates the Django permissions framework, but for API keys.

If we get this, I *think* we could also try to take inspiration from [`DjangoModelPermission`](https://github.com/encode/django-rest-framework/blob/master/rest_framework/permissions.py#L162). It features a `.params_map` class attribute which defines which permissions should be checked for each HTTP method. If we build an `APIKeyScopesPermission` permission class with a `.scopes_map` attribute, we'd be able to **control API key access at a higher level using _scopes_** instead of raw HTTP methods and endpoint paths.

And then maybe we can keep the notion of `APIKeyGroup` in order to mass-assign multiple scopes to an API key.

We'll then end up basically replicating the Django permissions system for API keys, but yeah, I think it's a good sign actually. :-)

How does this sound to you?